drivers/soundwire/master.c

```diff
@@ -21,50 +19,41 @@ struct device_type sdw_master_type = {
 
 /**
  * sdw_master_device_add() - create a Linux Master Device representation.
- * @parent: the parent Linux device (e.g. a PCI device)
- * @fwnode: the parent fwnode (e.g. an ACPI companion device to the parent)
- * @link_ops: link-specific ops (optional)
- * @link_id: link index as defined by MIPI DisCo specification
- * @pdata: private data (e.g. register base, offsets, platform quirks, etc).
- *
- * The link_ops argument can be NULL, it is only used when link-specific
- * initializations and power-management are required.
+ * @bus: SDW bus instance
+ * @parent: parent device
+ * @fwnode: firmware node handle
  */
-struct sdw_master_device
-*sdw_master_device_add(struct device *parent,
-		       struct fwnode_handle *fwnode,
-		       struct sdw_link_ops *link_ops,
-		       int link_id,
-		       void *pdata)
+int sdw_master_device_add(struct sdw_bus *bus, struct device *parent,
+			  struct fwnode_handle *fwnode)
 {
-	struct sdw_master_device *md;
+	struct sdw_master_device *md = &bus->md;
```

Comment:
Should we assign md after the ``if (!bus)`` test?

Reply:
yes indeed, tools would flag this, will fix.